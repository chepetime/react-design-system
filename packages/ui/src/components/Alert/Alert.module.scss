@use "./../../styles" as styles;

:root {
  --alert-background-color: #{styles.$pink-500};
}

:global :root .dark {
  --alert-background-color: #{styles.$indigo-500};
}

.Alert {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--alert-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
