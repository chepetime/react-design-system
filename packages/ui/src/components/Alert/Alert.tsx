import React from "react";
import classnames from "classnames";
import css from "./Alert.module.scss";

export interface AlertProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>(
  (props: AlertProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Alert)} ref={ref}>
        <p>Alert component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Alert.displayName = "Alert";
