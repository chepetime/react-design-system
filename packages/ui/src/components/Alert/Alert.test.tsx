import React from "react";
import { render } from "@testing-library/react";
import { Alert } from "./Alert";

describe("Alert", () => {
  test("renders the Alert component", () => {
    render(<Alert>This is an Alert Message</Alert>);
  });
});
