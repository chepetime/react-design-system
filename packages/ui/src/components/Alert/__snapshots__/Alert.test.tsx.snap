// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<div>
  <div
    class="Alert"
  >
    <p>
      Alert component working!
    </p>
    <div />
  </div>
</div>
`;
